client/layout/global-sidebar/menu-items/notifications/index.jsx

```diff
@@ -29,70 +25,32 @@ class SidebarNotifications extends Component {
 		onClick: PropTypes.func,
 		//connected
 		isNotificationsOpen: PropTypes.bool,
-		hasUnseenNotifications: PropTypes.bool,
+		unseenCount: PropTypes.number,
 		tooltip: TranslatableString,
 		translate: PropTypes.func,
 		currentUserId: PropTypes.number,
 		locale: PropTypes.string,
 	};
 
-	notificationLink = createRef();
-	notificationPanel = createRef();
-
 	state = {
 		animationState: 0,
-		newNote: this.props.hasUnseenNotifications,
+		newNote: this.props.unseenCount > 0,
```

Comment:
Was there a reason we stopped using the more robust selector for this? 
Relying on unseenCount alone caused a regression since it isn't initialized on calypso loading, while the old selector checks user data as a fallback.

Attempting to resolve here: https://github.com/Automattic/wp-calypso/pull/91308